fibonaci/fibonacci.py

```diff
@@ -0,0 +1,117 @@
+from decimal import Decimal, getcontext
+import math
+from functools import lru_cache
+from typing import Union, List
+
+__all__ = [
+    "fibonacci_closed",
+    "fibonacci_loop",
+    "fibonacci_range",
+    "fibonacci_loop_max",
+    "fibonacci_recursive",
+]
+
+
+def fibonacci_closed(n: int) -> int:
+    """Function that returns the nth number in the fibonacci sequence
+
+    Args:
+        n (int): Return the number of the fibonacci sequence of this n
+
+    Returns:
+        int: The nth number in the fibonacci sequence
+    """
+    getcontext().prec = 1000
+    phi = Decimal((1 + Decimal(5).sqrt()) / 2)
+    psi = Decimal((1 - Decimal(5).sqrt()) / 2)
+    return int((phi ** n - psi ** n) / Decimal(5).sqrt()+Decimal(0.5))
+
+
+def fibonacci_loop(n: int) -> Union[int, None]:
```

Comment:
`Optional[int]` does the same as `Union[int, None]`, and it is the preferred way to declare an optional return value. (or input parameter).